hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/staking/StakeRewardCalculatorImpl.java

```diff
@@ -14,64 +14,68 @@
  * limitations under the License.
  */
 
-package com.hedera.node.app.service.token.impl.staking;
+package com.hedera.node.app.service.token.impl.handlers.staking;
 
 import static com.hedera.node.app.service.token.Units.HBARS_TO_TINYBARS;
+import static com.hedera.node.app.service.token.impl.handlers.staking.StakingUtils.totalStake;
 
 import com.google.common.annotations.VisibleForTesting;
 import com.hedera.hapi.node.state.token.Account;
 import com.hedera.hapi.node.state.token.StakingNodeInfo;
-import com.hedera.node.app.service.mono.ledger.accounts.staking.StakePeriodManager;
+import com.hedera.node.app.service.token.ReadableNetworkStakingRewardsStore;
 import com.hedera.node.app.service.token.ReadableStakingInfoStore;
 import edu.umd.cs.findbugs.annotations.NonNull;
 import edu.umd.cs.findbugs.annotations.Nullable;
+import java.time.Instant;
 import java.util.List;
 import javax.inject.Inject;
 import javax.inject.Singleton;
 
 @Singleton
-public class RewardCalculator {
+public class StakeRewardCalculatorImpl implements StakeRewardCalculator {
     private final StakePeriodManager stakePeriodManager;
-    private long rewardsPaid;
 
     @Inject
-    public RewardCalculator(@NonNull final StakePeriodManager stakePeriodManager) {
+    public StakeRewardCalculatorImpl(@NonNull final StakePeriodManager stakePeriodManager) {
         this.stakePeriodManager = stakePeriodManager;
     }
 
-    public void reset() {
-        rewardsPaid = 0;
-    }
-
+    /** {@inheritDoc} */
+    @Override
     public long computePendingReward(
-            @NonNull final Account account, @NonNull final ReadableStakingInfoStore stakingInfoStore) {
+            @NonNull final Account account,
+            @NonNull final ReadableStakingInfoStore stakingInfoStore,
+            @NonNull final ReadableNetworkStakingRewardsStore rewardsStore,
+            @NonNull final Instant consensusNow) {
         final var effectiveStart = stakePeriodManager.effectivePeriod(account.stakePeriodStart());
-        if (!stakePeriodManager.isRewardable(effectiveStart)) {
+        if (!stakePeriodManager.isRewardable(effectiveStart, rewardsStore, consensusNow)) {
             return 0;
         }
 
-        final var addressBookId = calculateNodeAddressId(account.stakedNodeId());
-        final var stakingInfo = stakingInfoStore.get(addressBookId);
-        final var rewardOffered =
-                computeRewardFromDetails(account, stakingInfo, stakePeriodManager.currentStakePeriod(), effectiveStart);
+        final var nodeId = account.stakedNodeId();
+        final var stakingInfo = stakingInfoStore.get(nodeId);
```

Comment:
`mono-service` depends on using a single field for both staked node ids and staked account numbers.

But here we have a [`oneof staked_id`](https://github.com/hashgraph/hedera-protobufs/blob/add-pbj-types-for-state/services/state/token/account.proto#L84); and it must reference a node id at this point if the `StakingRewardsPayer` decided to call `computePendingReward()` for this account.

I think it would make sense for L55 to be,
```
final var nodeId = account.stakedNodeIdOrThrow();
```
...to emphasize this invariant.

Reply:
Agreed with @tinker-michaelj . Changed as suggested.

Reply:
Ahhhh, much easier. Thanks for explaining @tinker-michaelj!